jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleLayerConfigurations.java

```diff
@@ -132,5 +155,112 @@ static JavaLayerConfigurations getForProject(
     return layerBuilder.build();
   }
 
+  /**
+   * Resolves the {@link JavaLayerConfigurations} for a War Gradle {@link Project}.
+   *
+   * @param project the Gradle {@link Project}
+   * @param logger the build logger for providing feedback about the resolution
+   * @param extraDirectory path to the directory for the extra files layer
+   * @param appRoot root directory in the image where the app will be placed
+   * @return a {@link JavaLayerConfigurations} for the layers for the Gradle {@link Project}
+   * @throws IOException if an I/O exception occurred during resolution
+   */
+  private static JavaLayerConfigurations getForWarProject(
+      Project project, Logger logger, Path extraDirectory, AbsoluteUnixPath appRoot)
+      throws IOException {
+    List<Path> dependenciesFiles = new ArrayList<>();
+    List<Path> snapshotDependenciesFiles = new ArrayList<>();
+    List<Path> resourcesFiles = new ArrayList<>();
+    List<Path> resourcesWebInfFiles = new ArrayList<>();
+    List<Path> extraFiles = new ArrayList<>();
+
+    Path explodedWarPath = GradleProjectProperties.getExplodedWarDirectory(project);
+
+    Path libOutputDirectory =
+        explodedWarPath.resolve(JavaLayerConfigurations.WEB_INF_LIB_RELATIVE_PATH);
+    try (Stream<Path> dependencyFileStream = Files.list(libOutputDirectory)) {
+      dependencyFileStream.forEach(
+          path -> {
+            if (path.toString().contains(JavaLayerConfigurations.SNAPSHOT_FILENAME_SUFFIX)) {
+              snapshotDependenciesFiles.add(path);
+            } else {
+              dependenciesFiles.add(path);
+            }
+          });
+    }
+    // All files except classes and libs in resources
+    try (Stream<Path> fileStream = Files.list(explodedWarPath)) {
+      fileStream
+          .filter(path -> !path.endsWith(JavaLayerConfigurations.WEB_INF_RELATIVE_PATH))
+          .forEach(resourcesFiles::add);
+    }
+    // Some files in WEB-INF need to be in resources directory (e.g. web.xml, ...)
+    Path webinfOutputDirectory =
+        explodedWarPath.resolve(JavaLayerConfigurations.WEB_INF_RELATIVE_PATH);
+    try (Stream<Path> fileStream = Files.list(webinfOutputDirectory)) {
```

Comment:
We also need to check the directory existence for `WEB-INF`, `WEB-INF/lib`, and `WEB-INF/classes`. (See #1074.)

Reply:
Done